### ISSUE TITLE
macOS: segmentation fault 11

### ISSUE THREAD

**Issue by ambuehll** [eclipse/sumo#10334]
meso sim finishes, but throws on macOS: 
zsh: segmentation fault  sumo -c zuerich3.sumocfg 
[files](https://drive.google.com/drive/folders/1j5puPchGP7RA0lgI-IIelCvUCeni8CsU?usp=sharing)
runs without error on Windows
I saw https://github.com/eclipse/sumo/issues/6242#issuecomment-553458710, but I don't think it's related to that, because it does not happen in all scenarios. 

---
*If possible, upload an example that shows your problem.*

**SUMO-version:** 1.12.0 (Darwin-21.3.0 arm64 Clang 13.0.0.13000029 Release Proj GUI SWIG GDAL FFmpeg GL2PS)
built using homebrew 
**operating system:** macOS Monterey (M1 Max)

**Comment by namdre** [comment]
does it also crash with the latest dev version? Several crashes have been found since release.

**Comment by ambuehll** [comment]
I tried installing the newest version using pip on mac. 
ERROR: Could not find a version that satisfies the requirement eclipse-sumo (from versions: none)
Not working for now. Will have to build it from scratch - I'll do this tonight and come back again.

**Comment by ambuehll** [comment]
Ran it with current dev v1_12_0+1381-802d85a9a91 and I am getting the same warning/error:
`zsh: segmentation fault  sumo -c zuerich3.sumocfg
`

**Comment by namdre** [comment]
Simulation is non-deterministic due to #10292
Does it crash without option --device.rerouting.threads ?

**Comment by ambuehll** [comment]
only in the parallel routing case. 
no crash without device.rerouting.threads.

**Comment by namdre** [comment]
Due to non-determinism, the runs with parallel routing explore a larger configuration space and can therefore trigger a crash that the deterministic run is missing.
My best chance for fixing the bug is to catch it in a deterministic simulation (unless is actually rooted in the parallelism).
The following could help:

Run the non-parallel scenario with as many different seeds as you can (option --seed) and report a crashing seed if you find one.

Side note:  This could be simplified by running it many times with option --random but due to #7062 we cannot currently reproduce those runs so it wouldn't be as useful.

**Comment by ambuehll** [comment]
I checked 50 random seeds in non-parallel. None crashed. 
I checked 10 random seeds with `device-reroute.threads 7`; all of them crashed.
When running it with less than 6 threads, it doesn't crash. Crashes with 6-9 (haven't tested anything higher given the number of cores I have). 

NB: I have other scenarios that fail with 4 threads already (I can also provide them if helpful).

**Comment by namdre** [comment]
I'm trying to reproduce the crash on Linux which is my main development platform but haven't been able to do so. The fact that the Zurich-scenarion only crashes with a high number of threads is unlike any other parallelization-related crash that I've come across thus far. Honestly, I'm not sure how to best proceed from here.
Could you build sumo in debug mode, run the scenario in gdb and attach stack traces of the different threads?

**Comment by ambuehll** [comment]
First, thank you for all the help! very much appreciated. 
Second, I am not an expert for the debug mode but I'll give it a try later today.

**Comment by ambuehll** [comment]
just found out that it only crashes when `device.rerouting.with-taz false`, not however, when it is true.

**Comment by ambuehll** [comment]
Building sumo using `gcc` instead of `clang` solves the problem.

**Comment by ambuehll** [comment]
Finally came around to run sumo in debug mode with lldb. Hope this was what you were looking for. The issue persists on every mac I came across (4x). So I am going to re-open this issue, if that's OK.

Let me know if you need more info. Files will follow shortly.
Thanks!
Lukas

```
(lldb) target create "/Users/ambuehll/sumo/bin/sumoD"
Current executable set to '/Users/ambuehll/sumo/bin/sumoD' (arm64).
(lldb) run -c /Users/ambuehll/transcality/astra_rapp_output/eODs/sumo.sumocfg
Process 51187 launched: '/Users/ambuehll/sumo/bin/sumoD' (arm64)
Warning: Could not set locale to 'C'.
Process 51187 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x10)
    frame #0: 0x0000000103f96f5c libstdc++.6.dylib`std::_Rb_tree_rebalance_for_erase(std::_Rb_tree_node_base*, std::_Rb_tree_node_base&) + 284
libstdc++.6.dylib`std::_Rb_tree_rebalance_for_erase:
->  0x103f96f5c <+284>: ldr    x5, [x3, #0x10]
    0x103f96f60 <+288>: cbz    x5, 0x103f96f70           ; <+304>
    0x103f96f64 <+292>: ldr    w6, [x5]
    0x103f96f68 <+296>: cmp    w6, #0x1
Target 0: (sumoD) stopped.
(lldb) bt all
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x10)
  * frame #0: 0x0000000103f96f5c libstdc++.6.dylib`std::_Rb_tree_rebalance_for_erase(std::_Rb_tree_node_base*, std::_Rb_tree_node_base&) + 284
    frame #1: 0x000000010007c084 sumoD`std::_Rb_tree<SUMOVehicle const*, SUMOVehicle const*, std::_Identity<SUMOVehicle const*>, std::less<SUMOVehicle const*>, std::allocator<SUMOVehicle const*> >::_M_erase_aux(this=0x000000010503b870, __position=_Rb_tree<const SUMOVehicle *, const SUMOVehicle *, _Identity<const SUMOVehicle *>, less<const SUMOVehicle *>, allocator<> >::const_iterator @ 0x000000016fdfeeb0) at stl_tree.h:2493:5
    frame #2: 0x000000010007adcc sumoD`std::_Rb_tree<SUMOVehicle const*, SUMOVehicle const*, std::_Identity<SUMOVehicle const*>, std::less<SUMOVehicle const*>, std::allocator<SUMOVehicle const*> >::_M_erase_aux(this=0x000000010503b870, __first=_Rb_tree<const SUMOVehicle *, const SUMOVehicle *, _Identity<const SUMOVehicle *>, less<const SUMOVehicle *>, allocator<> >::const_iterator @ 0x000000016fdfeef0, __last=_Rb_tree<const SUMOVehicle *, const SUMOVehicle *, _Identity<const SUMOVehicle *>, less<const SUMOVehicle *>, allocator<> >::const_iterator @ 0x000000016fdfeee8) at stl_tree.h:2509:16
    frame #3: 0x0000000100079c2c sumoD`std::_Rb_tree<SUMOVehicle const*, SUMOVehicle const*, std::_Identity<SUMOVehicle const*>, std::less<SUMOVehicle const*>, std::allocator<SUMOVehicle const*> >::erase(this=0x000000010503b870, __x=0x000000016fdfefa8) at stl_tree.h:2520:19
    frame #4: 0x0000000100078ce8 sumoD`std::set<SUMOVehicle const*, std::less<SUMOVehicle const*>, std::allocator<SUMOVehicle const*> >::erase(this=0x000000010503b870, __x=0x000000016fdfefa8) at stl_set.h:685:26
    frame #5: 0x0000000100076e74 sumoD`MSInsertionControl::tryInsert(this=0x000000010503b800, time=865000, veh=0x0000000296cc3e10, refusedEmits=0x000000016fdff008) at MSInsertionControl.cpp:169:29
    frame #6: 0x0000000100076b8c sumoD`MSInsertionControl::emitVehicles(this=0x000000010503b800, time=865000) at MSInsertionControl.cpp:139:36
    frame #7: 0x00000001000db884 sumoD`MSNet::simulationStep(this=0x000000010503b200) at MSNet.cpp:784:29
    frame #8: 0x00000001000d6f18 sumoD`MSNet::simulate(this=0x000000010503b200, start=0, stop=3600000) at MSNet.cpp:400:23
    frame #9: 0x000000010000b540 sumoD`main(argc=3, argv=0x000000016fdff7b0) at sumo_main.cpp:99:38
    frame #10: 0x0000000190cc7e50 dyld`start + 2544
  thread #2
    frame #0: 0x0000000190fb3a1c libsystem_kernel.dylib`__workq_kernreturn + 8
  thread #4
    frame #0: 0x0000000190fb5564 libsystem_kernel.dylib`__psynch_cvwait + 8
    frame #1: 0x0000000190ff1638 libsystem_pthread.dylib`_pthread_cond_wait + 1232
    frame #2: 0x000000010003a868 sumoD`MFXWorkerThread::run(this=0x0000000130beb750) at MFXWorkerThread.h:340:33
    frame #3: 0x0000000103a3ba6c libFOX-1.6.0.dylib`FX::FXThread::execute(void*) + 68
    frame #4: 0x0000000190ff106c libsystem_pthread.dylib`_pthread_start + 148
  thread #5
    frame #0: 0x0000000190fb5564 libsystem_kernel.dylib`__psynch_cvwait + 8
    frame #1: 0x0000000190ff1638 libsystem_pthread.dylib`_pthread_cond_wait + 1232
    frame #2: 0x000000010003a868 sumoD`MFXWorkerThread::run(this=0x0000000130beb8d0) at MFXWorkerThread.h:340:33
    frame #3: 0x0000000103a3ba6c libFOX-1.6.0.dylib`FX::FXThread::execute(void*) + 68
    frame #4: 0x0000000190ff106c libsystem_pthread.dylib`_pthread_start + 148
  thread #6
    frame #0: 0x0000000190fb5564 libsystem_kernel.dylib`__psynch_cvwait + 8
    frame #1: 0x0000000190ff1638 libsystem_pthread.dylib`_pthread_cond_wait + 1232
    frame #2: 0x000000010003a868 sumoD`MFXWorkerThread::run(this=0x0000000130beba50) at MFXWorkerThread.h:340:33
    frame #3: 0x0000000103a3ba6c libFOX-1.6.0.dylib`FX::FXThread::execute(void*) + 68
    frame #4: 0x0000000190ff106c libsystem_pthread.dylib`_pthread_start + 148
  thread #7
    frame #0: 0x0000000190fb5564 libsystem_kernel.dylib`__psynch_cvwait + 8
    frame #1: 0x0000000190ff1638 libsystem_pthread.dylib`_pthread_cond_wait + 1232
    frame #2: 0x000000010003a868 sumoD`MFXWorkerThread::run(this=0x0000000117cb7e50) at MFXWorkerThread.h:340:33
    frame #3: 0x0000000103a3ba6c libFOX-1.6.0.dylib`FX::FXThread::execute(void*) + 68
    frame #4: 0x0000000190ff106c libsystem_pthread.dylib`_pthread_start + 148
  thread #8
    frame #0: 0x0000000190fb5564 libsystem_kernel.dylib`__psynch_cvwait + 8
    frame #1: 0x0000000190ff1638 libsystem_pthread.dylib`_pthread_cond_wait + 1232
    frame #2: 0x000000010003a868 sumoD`MFXWorkerThread::run(this=0x0000000117cb7fd0) at MFXWorkerThread.h:340:33
    frame #3: 0x0000000103a3ba6c libFOX-1.6.0.dylib`FX::FXThread::execute(void*) + 68
    frame #4: 0x0000000190ff106c libsystem_pthread.dylib`_pthread_start + 148

```

**Comment by therealthingy** [comment]
I'm getting (presumably) the same crash on Ubuntu 22.04 LTS:

```
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `sumo -c osm.sumocfg --threads 48 --no-warnings'.
bProgram terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f37e18c6aa4 in std::_Rb_tree_rebalance_for_erase(std::_Rb_tree_node_base*, std::_Rb_tree_node_base&) () from /lib/x86_64-linux-gnu/libstdc++.so.6
[Current thread is 1 (Thread 0x7f37dbe34640 (LWP 30356))]
gef➤  bt
#0  0x00007f37e18c6aa4 in std::_Rb_tree_rebalance_for_erase(std::_Rb_tree_node_base*, std::_Rb_tree_node_base&) () from /lib/x86_64-linux-gnu/libstdc++.so.6
#1  0x000055e3717327b1 in ?? ()
#2  0x000055e3716c492f in ?? ()
#3  0x000055e3716af57c in ?? ()
#4  0x000055e3716dd3c5 in ?? ()
#5  0x000055e3717146fb in ?? ()
#6  0x000055e371704f30 in ?? ()
#7  0x000055e371705575 in ?? ()
#8  0x00007f37e1e624f3 in FX::FXThread::execute(void*) () from /lib/x86_64-linux-gnu/libFOX-1.6.so.0
#9  0x00007f37e1494b43 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#10 0x00007f37e1526a00 in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
gef➤  i sharedlibrary
From                To                  Syms Read   Shared Object Library
0x00007f37e24e2cd0  0x00007f37e2653521  Yes (*)     /lib/x86_64-linux-gnu/libxerces-c-3.2.so
0x00007f37e275f280  0x00007f37e276fc14  Yes (*)     /lib/x86_64-linux-gnu/libz.so.1
0x00007f37e2069eb0  0x00007f37e22b6c80  Yes (*)     /lib/x86_64-linux-gnu/libproj.so.22
0x00007f37e1d2e920  0x00007f37e1e95850  Yes (*)     /lib/x86_64-linux-gnu/libFOX-1.6.so.0
0x00007f37e18a2440  0x00007f37e19a9b02  Yes (*)     /lib/x86_64-linux-gnu/libstdc++.so.6
0x00007f37e1b273a0  0x00007f37e1ba28e8  Yes         /lib/x86_64-linux-gnu/libm.so.6
0x00007f37e23e3660  0x00007f37e23f9be5  Yes (*)     /lib/x86_64-linux-gnu/libgcc_s.so.1
0x00007f37e1428700  0x00007f37e15baabd  Yes         /lib/x86_64-linux-gnu/libc.so.6
0x00007f37e1a87d00  0x00007f37e1af6291  Yes (*)     /lib/x86_64-linux-gnu/libcurl-gnutls.so.4
0x00007f37e126b910  0x00007f37e135de31  Yes (*)     /lib/x86_64-linux-gnu/libicuuc.so.70
0x00007f37e16d1600  0x00007f37e17bc32d  Yes (*)     /lib/x86_64-linux-gnu/libsqlite3.so.0
0x00007f37e1635f60  0x00007f37e167c6b8  Yes (*)     /lib/x86_64-linux-gnu/libtiff.so.5
0x00007f37e10df0c0  0x00007f37e11695ee  Yes (*)     /lib/x86_64-linux-gnu/libX11.so.6
0x00007f37e23cf600  0x00007f37e23d994e  Yes (*)     /lib/x86_64-linux-gnu/libXext.so.6
0x00007f37e23b6f80  0x00007f37e23c4b53  Yes (*)     /lib/x86_64-linux-gnu/libXft.so.2
0x00007f37e23a8660  0x00007f37e23ad1fc  Yes (*)     /lib/x86_64-linux-gnu/libXcursor.so.1
0x00007f37e1fbdfa0  0x00007f37e1fe5b1e  Yes (*)     /lib/x86_64-linux-gnu/libfontconfig.so.1
0x00007f37e100a7b0  0x00007f37e109259c  Yes (*)     /lib/x86_64-linux-gnu/libfreetype.so.6
0x00007f37e23a2040  0x00007f37e23a2105  Yes         /lib/x86_64-linux-gnu/libdl.so.2
0x00007f37e239d040  0x00007f37e239d105  Yes         /lib/x86_64-linux-gnu/libpthread.so.0
0x00007f37e0f80540  0x00007f37e0fc8861  Yes (*)     /lib/x86_64-linux-gnu/libjpeg.so.8
0x00007f37e1a41530  0x00007f37e1a69479  Yes (*)     /lib/x86_64-linux-gnu/libpng16.so.16
0x00007f37e0f2a7a0  0x00007f37e0f6bc3d  Yes (*)     /lib/x86_64-linux-gnu/libGLU.so.1
0x00007f37e0ee21c0  0x00007f37e0ee4f11  Yes (*)     /lib/x86_64-linux-gnu/libGL.so.1
0x00007f37e2794090  0x00007f37e27bd335  Yes         /lib64/ld-linux-x86-64.so.2
0x00007f37e23771c0  0x00007f37e238b546  Yes (*)     /lib/x86_64-linux-gnu/libnghttp2.so.14
0x00007f37e1f973e0  0x00007f37e1f9a579  Yes (*)     /lib/x86_64-linux-gnu/libidn2.so.0
0x00007f37e0e85d80  0x00007f37e0e94fd9  Yes (*)     /lib/x86_64-linux-gnu/librtmp.so.1
0x00007f37e0e22460  0x00007f37e0e62698  Yes (*)     /lib/x86_64-linux-gnu/libssh.so.4
0x00007f37e0e013e0  0x00007f37e0e02ea3  Yes (*)     /lib/x86_64-linux-gnu/libpsl.so.5
0x00007f37e0dc3260  0x00007f37e0de57dc  Yes (*)     /lib/x86_64-linux-gnu/libnettle.so.8
0x00007f37e0c01b40  0x00007f37e0d27afd  Yes (*)     /lib/x86_64-linux-gnu/libgnutls.so.30
0x00007f37e0b873f0  0x00007f37e0bbc8c8  Yes (*)     /lib/x86_64-linux-gnu/libgssapi_krb5.so.2
0x00007f37e0b2b000  0x00007f37e0b62ddc  Yes (*)     /lib/x86_64-linux-gnu/libldap-2.5.so.0
0x00007f37e1a2e360  0x00007f37e1a35341  Yes (*)     /lib/x86_64-linux-gnu/liblber-2.5.so.0
0x00007f37e0a56220  0x00007f37e0b07f02  Yes (*)     /lib/x86_64-linux-gnu/libzstd.so.1
0x00007f37e0a3f140  0x00007f37e0a460da  Yes (*)     /lib/x86_64-linux-gnu/libbrotlidec.so.1
0x00007f37dee01050  0x00007f37dee011a1  Yes (*)     /lib/x86_64-linux-gnu/libicudata.so.70
0x00007f37ded96300  0x00007f37dedead3d  Yes (*)     /lib/x86_64-linux-gnu/libwebp.so.7
0x00007f37ded6c3c0  0x00007f37ded860de  Yes (*)     /lib/x86_64-linux-gnu/liblzma.so.5
0x00007f37e0a2f140  0x00007f37e0a37979  Yes (*)     /lib/x86_64-linux-gnu/libjbig.so.0
0x00007f37ded460a0  0x00007f37ded61de6  Yes (*)     /lib/x86_64-linux-gnu/libdeflate.so.0
0x00007f37ded26640  0x00007f37ded39660  Yes (*)     /lib/x86_64-linux-gnu/libxcb.so.1
0x00007f37e0a22440  0x00007f37e0a287e2  Yes (*)     /lib/x86_64-linux-gnu/libXrender.so.1
0x00007f37e1f8f310  0x00007f37e1f91961  Yes (*)     /lib/x86_64-linux-gnu/libXfixes.so.3
0x00007f37decee290  0x00007f37ded0c5a7  Yes (*)     /lib/x86_64-linux-gnu/libexpat.so.1
0x00007f37dece3550  0x00007f37dece6d61  Yes (*)     /lib/x86_64-linux-gnu/libuuid.so.1
0x00007f37decca1a0  0x00007f37decca60b  Yes (*)     /lib/x86_64-linux-gnu/libOpenGL.so.0
0x00007f37dec3d240  0x00007f37dec3f187  Yes (*)     /lib/x86_64-linux-gnu/libGLdispatch.so.0
0x00007f37debcc6c0  0x00007f37debe6080  Yes (*)     /lib/x86_64-linux-gnu/libGLX.so.0
0x00007f37dea30790  0x00007f37dea65b51  Yes (*)     /lib/x86_64-linux-gnu/libunistring.so.2
0x00007f37de9dfc80  0x00007f37de9f1b89  Yes (*)     /lib/x86_64-linux-gnu/libhogweed.so.6
0x00007f37de95f440  0x00007f37de9bd54d  Yes (*)     /lib/x86_64-linux-gnu/libgmp.so.10
0x00007f37de4b4000  0x00007f37de70ee22  Yes (*)     /lib/x86_64-linux-gnu/libcrypto.so.3
0x00007f37de2eebe0  0x00007f37de3913a0  Yes (*)     /lib/x86_64-linux-gnu/libp11-kit.so.0
0x00007f37de9404a0  0x00007f37de94d20b  Yes (*)     /lib/x86_64-linux-gnu/libtasn1.so.6
0x00007f37de894f50  0x00007f37de8efb77  Yes (*)     /lib/x86_64-linux-gnu/libkrb5.so.3
0x00007f37de8474c0  0x00007f37de8615f2  Yes (*)     /lib/x86_64-linux-gnu/libk5crypto.so.3
0x00007f37de2c13c0  0x00007f37de2c1f59  Yes (*)     /lib/x86_64-linux-gnu/libcom_err.so.2
0x00007f37de2b4630  0x00007f37de2b9a24  Yes (*)     /lib/x86_64-linux-gnu/libkrb5support.so.0
0x00007f37de299860  0x00007f37de2a90e4  Yes (*)     /lib/x86_64-linux-gnu/libsasl2.so.2
0x00007f37de274080  0x00007f37de2745b5  Yes (*)     /lib/x86_64-linux-gnu/libbrotlicommon.so.1
0x00007f37de26e360  0x00007f37de26f092  Yes (*)     /lib/x86_64-linux-gnu/libXau.so.6
0x00007f37de2671a0  0x00007f37de268967  Yes (*)     /lib/x86_64-linux-gnu/libXdmcp.so.6
0x00007f37de25a460  0x00007f37de260392  Yes (*)     /lib/x86_64-linux-gnu/libffi.so.8
0x00007f37de253260  0x00007f37de254404  Yes (*)     /lib/x86_64-linux-gnu/libkeyutils.so.1
0x00007f37de2406a0  0x00007f37de249229  Yes         /lib/x86_64-linux-gnu/libresolv.so.2
0x00007f37de22a040  0x00007f37de235226  Yes (*)     /lib/x86_64-linux-gnu/libbsd.so.0
0x00007f37de21a160  0x00007f37de220739  Yes (*)     /lib/x86_64-linux-gnu/libmd.so.0
(*): Shared library is missing debugging information.
```


The issue hence seems to be not limited to macOS

**Comment by namdre** [comment]
@therealthingy
- can you try to get a stack trace with a debug-build of sumo?
- are you only seeing the crash when setting option **--threads**?

note, that parallelization via option **--threads** currently gives a very low speedup. If you have to run more than one simulation, your compute is better spent running multiple simulations in parallel rather than trying to run one simulation very fast.

**Comment by therealthingy** [comment]
@namdre
Sorry for the belated response (I was busy 😅).


In my case it seems to be a heap memory corruption issue, which is linked to the threads option (as shown down below).
And no, during my **brief** testing, the issue didn't occur when running w/ a single thread.
This might therefore be a separate issue (unrelated to @ambuehll's issue).

```
❯ sumo -c osm.sumocfg --threads 24 --no-warnings
Warning: Environment variable SUMO_HOME is not set, schema resolution will use slow website lookups.
Loading net-file from 'osm.net.xml' ... done (1820ms).
Loading additional-files from 'vehicleTypes.xml' ... done (1ms).
Loading additional-files from 'routesPKW.xml' ... done (211ms).
Loading additional-files from 'routesLKW.xml' ... done (9ms).
Loading additional-files from 'osm.poly.xml' ... done (55ms).
Loading done.
Simulation version 1.12.0 started with time: 0.00
free(): invalid pointer
[1]    3927 IOT instruction (core dumped)  sumo -c osm.sumocfg --threads 24 --no-warnings

❯ gdb $(which sumo) core.3927
GNU gdb (Ubuntu 12.1-0ubuntu1~22.04) 12.1
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
GEF for linux ready, type `gef' to start, `gef config' to configure
90 commands loaded and 5 functions added for GDB 12.1 in 0.00ms using Python engine 3.10
Reading symbols from /usr/bin/sumo...
(No debugging symbols found in /usr/bin/sumo)
[New LWP 3946]
[New LWP 3939]
[New LWP 3938]
[New LWP 3927]
[New LWP 3941]
[New LWP 3940]
[New LWP 3943]
[New LWP 3945]
[New LWP 3944]
[New LWP 3947]
[New LWP 3948]
[New LWP 3951]
[New LWP 3942]
[New LWP 3949]
[New LWP 3959]
[New LWP 3952]
[New LWP 3957]
[New LWP 3955]
[New LWP 3953]
[New LWP 3958]
[New LWP 3956]
[New LWP 3950]
[New LWP 3960]
[New LWP 3954]
[New LWP 3961]
[*] Failed to find objfile or not a valid file format: [Errno 2] No such file or directory: 'system-supplied DSO at 0x7ffdb8ef7000'
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `sumo -c osm.sumocfg --threads 24 --no-warnings'.
Program terminated with signal SIGABRT, Aborted.
#0  __pthread_kill_implementation (no_tid=0x0, signo=0x6, threadid=0x7f8b1526f640) at ./nptl/pthread_kill.c:44
44	./nptl/pthread_kill.c: No such file or directory.
[Current thread is 1 (Thread 0x7f8b1526f640 (LWP 3946))]
gef➤  bt
#0  __pthread_kill_implementation (no_tid=0x0, signo=0x6, threadid=0x7f8b1526f640) at ./nptl/pthread_kill.c:44
#1  __pthread_kill_internal (signo=0x6, threadid=0x7f8b1526f640) at ./nptl/pthread_kill.c:78
#2  __GI___pthread_kill (threadid=0x7f8b1526f640, signo=signo@entry=0x6) at ./nptl/pthread_kill.c:89
#3  0x00007f8b1c842476 in __GI_raise (sig=sig@entry=0x6) at ../sysdeps/posix/raise.c:26
#4  0x00007f8b1c8287f3 in __GI_abort () at ./stdlib/abort.c:79
#5  0x00007f8b1c8896f6 in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7f8b1c9dbb8c "%s\n") at ../sysdeps/posix/libc_fatal.c:155
#6  0x00007f8b1c8a0d7c in malloc_printerr (str=str@entry=0x7f8b1c9d9764 "free(): invalid pointer") at ./malloc/malloc.c:5664
#7  0x00007f8b1c8a2ac4 in _int_free (av=<optimized out>, p=<optimized out>, have_lock=0x0) at ./malloc/malloc.c:4439
#8  0x00007f8b1c8a54d3 in __GI___libc_free (mem=<optimized out>) at ./malloc/malloc.c:3391
#9  0x000055b85cee67b9 in ?? ()
#10 0x000055b85ce7892f in ?? ()
#11 0x000055b85ce6357c in ?? ()
#12 0x000055b85ce913c5 in ?? ()
#13 0x000055b85cec86fb in ?? ()
#14 0x000055b85ceb8f30 in ?? ()
#15 0x000055b85ceb9575 in ?? ()
#16 0x00007f8b1d2624f3 in FX::FXThread::execute(void*) () from /lib/x86_64-linux-gnu/libFOX-1.6.so.0
#17 0x00007f8b1c894b43 in start_thread (arg=<optimized out>) at ./nptl/pthread_create.c:442
#18 0x00007f8b1c926a00 in clone3 () at ../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
```

The segfault also seems to be related to this issue.
```
❯ sumo -c osm.sumocfg --threads 24 --no-warnings
Warning: Environment variable SUMO_HOME is not set, schema resolution will use slow website lookups.
Loading net-file from 'osm.net.xml' ... done (1803ms).
Loading additional-files from 'vehicleTypes.xml' ... done (0ms).
Loading additional-files from 'routesPKW.xml' ... done (212ms).
Loading additional-files from 'routesLKW.xml' ... done (9ms).
Loading additional-files from 'osm.poly.xml' ... done (52ms).
Loading done.
Simulation version 1.12.0 started with time: 0.00
[1]    3599 segmentation fault  sumo -c osm.sumocfg --threads 24 --no-warnings
```

Regarding the debug build: I've tried building it in run-level 3 (to conserve memory), but to no avail. The build always failed w/ `Out of memory: Killed process 3690 (cc1plus) total-vm:227700kB, anon-rss: 158460KB, file-rss:2560kB, shmem-rss:OKB, …` 
32GB aren't sufficient for compiling a debug build 😅.

**Comment by behrisch** [comment]
@ambuehll Is this still a problem?

**Comment by behrisch** [comment]
@therealthingy Your problem is probably independent because it is about parallelized simulation (not routing) which is in a very experimental stage anyway. If the problem persists, can you share your scenario (maybe in a new issue)?

**Comment by ambuehll** [comment]
hello, yes, with version 1_18_0+1373-99e33d5ed24.
What I did: downloaded the initial data from this issue and ran it twice with a different random seed: 1 ran through, the other threw a segmentation fault. 
@MatteoFel also has this issue. 
Anything we can test to help?

**Comment by behrisch** [comment]
Your comment above said "files follow shortly". Did you send them to Jakob directly or did this get lost somehow?

**Comment by ambuehll** [comment]
Not sure I remember, but I think it's more likely that they got lost. I can see if I can re-run the lldb if you'd like. The test I did yesterday was using these files: 

> meso sim finishes, but throws on macOS: zsh: segmentation fault sumo -c zuerich3.sumocfg [files](https://drive.google.com/drive/folders/1j5puPchGP7RA0lgI-IIelCvUCeni8CsU?usp=sharing)
